cmd/disk-cache.go

```diff
@@ -229,11 +132,24 @@ func (c cacheObjects) GetObjectNInfo(ctx context.Context, bucket, object string,
 	// Since we got here, we are serving the request from backend,
 	// and also adding the object to the cache.
 
-	if rs != nil {
-		// We don't cache partial objects.
+	if !dcache.diskAvailable(objInfo.Size) {
 		return c.GetObjectNInfoFn(ctx, bucket, object, rs, h, lockType, opts)
 	}
-	if !dcache.diskAvailable(objInfo.Size) {
+
+	if rs != nil {
```

Comment:
lot of parallel range gets will cause multiple background GETs of full objects.

If you want to fix it you can fix that cleanly in another PR.

Reply:
parallel range GETs will still cache only one entry - keeping a lock at the higher layer and checking if object is already cached will prevent multiple attempts to cache the object.